### PR TITLE
dev-util/clion: version bump to 2018.1.5

### PR DESCRIPTION
**Pull request by strahlc**
Bug: https://bugs.gentoo.org/653952
- dev-util/clion: version bump to 2018.1.5
- dev-util/clion: remove old version

### CODE REVIEW

**Comment by strahlc** [comment]
@mgorny I need help for resolving https://bugs.gentoo.org/652040 on this issue. Unfortunately I'm not a licensing expert.

**Comment by mgorny** [comment]
Well, the biggest part is that you need to look through all installed files, and find their licenses. This bundles a lot of stuff, so you basically need to look all those files, and either find licenses inside, or see what the library is and find license on its homepage.

File: dev-util/clion/clion-2017.3.4.ebuild [hunk 1]
```diff
@@ -10,7 +10,10 @@ HOMEPAGE="http://www.jetbrains.com/clion"
 SRC_URI="http://download.jetbrains.com/cpp/CLion-${PV}.tar.gz -> ${P}.tar.gz"
 
 LICENSE="IDEA
```
Comment (mgorny) [inline dev-util/clion/clion-2017.3.4.ebuild]: Do you really need to unconditionally agree to commercial license when you want to use one of the licenses below?

Comment (strahlc) [reply dev-util/clion/clion-2017.3.4.ebuild]: I don't know. I couldn't find a license for CLion. Neither on JetBrains website nor in the downloaded tar.gz from CLion. All licenses shipped with the tar.gz are related to components used in or bundled with CLion.

But I've found out that CLion doesn't has a community version that is free to use for open source development, like IDEA has. Thats the reason I've removed the IDEA_OpenSource license.

At work I've got a license key from JetBrains to use CLion, but no file with the license text.

**Comment by mgorny** [comment]
@gentoo/licenses, could you take a look?

**Comment by ulm** [comment]
AFAICS, all the IDEA* licenses are EULAs, by the language used in the first paragraph.
Furthermore, section 4.(b)(i) explicitly forbids redistribution.

So all ebuilds need mirror and bindist restriction, and the licenses should be added to the @EULA group.

About bundled software, I see at least the following licenses there:
Apache-2.0 Apache-1.1 GPL-3+ BSD BSD-2 MIT HPND OFL-1.1 PSF-2 EPL-1.0 JDOM ZLIB
Furthermore, there is a bunch of "all rights reserved", a "WAP Copyright Licence", and an "Open Mobile Alliance Use Agreement".

Note that the above is *not* a complete license audit. Also, I went through about one quarter of the tarball's content only.

**Comment by strahlc** [comment]
These are the licenses I've found:
```
$ equery f clion | grep -i license
/opt/clion-2018.1.3/jre64/LICENSE						GPL-2
/opt/clion-2018.1.3/jre64/lib/fonts/LICENSE.txt				Apache-2.0
/opt/clion-2018.1.3/license/GDB-license.txt				GPL-3
/opt/clion-2018.1.3/license/GDB-sources.txt				GPL-3
/opt/clion-2018.1.3/license/LLDB-license.txt				UoI-NCSA
/opt/clion-2018.1.3/license/Python-license.txt				PSF-2
/opt/clion-2018.1.3/license/XStream_license.txt				BSD
/opt/clion-2018.1.3/license/ant_license.txt				Apache-2.0
/opt/clion-2018.1.3/license/asm_license.txt				BSD
/opt/clion-2018.1.3/license/eclipse_license.txt				EPL-1.0
/opt/clion-2018.1.3/license/eclipse_license2.txt			EPL-2.0
/opt/clion-2018.1.3/license/growl.license					BSD
/opt/clion-2018.1.3/license/gson_license.txt				Apache-2.0
/opt/clion-2018.1.3/license/imgscalr_license.txt			Apache-2.0
/opt/clion-2018.1.3/license/javahelp_license.html			sun-bcla-javahelp
/opt/clion-2018.1.3/license/javolution_license.txt			BSD-2
/opt/clion-2018.1.3/license/jaxb_license.txt				CDDL-1.1 GPL-2-with-classpath-exception
/opt/clion-2018.1.3/license/jaxen_license.txt				BSD
/opt/clion-2018.1.3/license/jayatana_license.txt			MIT
/opt/clion-2018.1.3/license/jdom_license.txt				JDOM
/opt/clion-2018.1.3/license/jgoodies_forms_license.txt			BSD
/opt/clion-2018.1.3/license/junit_license.txt				CPL-0.5
/opt/clion-2018.1.3/license/kryo-license.txt				BSD
/opt/clion-2018.1.3/license/libstdcxx_printers-license.txt		GPL-3
/opt/clion-2018.1.3/license/libstdcxx_printers-sources.txt		GPL-3
/opt/clion-2018.1.3/license/log4j_license.txt				Apache-1.1
/opt/clion-2018.1.3/license/microba_license.txt				BSD
/opt/clion-2018.1.3/license/miglayout_swing_license.txt		BSD
/opt/clion-2018.1.3/license/nanoxml_license.txt				ZLIB
/opt/clion-2018.1.3/license/oromatcher_license.txt			oromatcher
/opt/clion-2018.1.3/license/picoContainer_license.txt			BSD
/opt/clion-2018.1.3/license/saxon-conditions.html			MPL-1.0
/opt/clion-2018.1.3/license/swingx_license.txt				LGPL-2.1+
/opt/clion-2018.1.3/license/trove4j_license.txt				LGPL-2.1+
/opt/clion-2018.1.3/license/winp_license.txt				MIT
/opt/clion-2018.1.3/license/xerces_license.txt				Apache-2.0
/opt/clion-2018.1.3/license/xmlrpc_license.txt				Apache-2.0
/opt/clion-2018.1.3/license/yourkit-license-redist.txt			BSD
/opt/clion-2018.1.3/plugins/JavaScriptLanguage/helpers/node_modules/semver/LICENSE		ISC
/opt/clion-2018.1.3/plugins/python/helpers/pydev/LICENSE					EPL-1.0
/opt/clion-2018.1.3/plugins/python/helpers/python-skeletons/LICENSE.txt			Apache-2.0
/opt/clion-2018.1.3/plugins/python/helpers/typeshed/LICENSE					Apache-2.0
/opt/clion-2018.1.3/plugins/tfsIntegration/lib/com.microsoft.tfs.sdk-14.0.1_license.html		ms-tfs-sdk
```

From that I derived the following license string (for the bundled stuff only):
```
LICENSE="GPL-2 Apache-2.0 GPL-3 UoI-NCSA PSF-2 BSD EPL-1.0 EPL-2.0 sun-bcla-javahelp BSD-2 CDDL-1.1 GPL-2-with-classpath-exception JDOM CPL-0.5 Apache-1.1 ZLIB oromatcher MPL-1.0 LGPL-2.1+ MIT ISC ms-tfs-sdk"
```

Four of the licenses are not in portage at the moment, but they'll be added with this merge request:
```
CDDL-1.1
oromatcher
ms-tfs-sdk
sun-bcla-javahelp
```

I didn't find any other licenses in this package, but on the website (https://www.jetbrains.com/clion/buy/#edition=discounts) I can see that one can use CLion with a commercial license, a personal license and 11 forms of discounted licenses (classroom, student, teacher, open source, startups, ...).
But I couldn't find license files for all of them. Furthermore these licenses are not CLion specific, they apply to the "JetBrains Toolbox Subscription" which contains a lot of tools (CLion, PyCharm, IntelliJ IDEA, WebStorm, GoLand, PHPStorm, ...).
I think that most of the licenses have to be listed in all JetBrains products (also PyCharm, IntelliJ IDEA, WebStorm, ...). For that reason I would like to split the version bump (the two first commits of this merge request) from the licensing topic. 
@mgorny: What do you think about that?

**Comment by strahlc** [comment]
@gentoo/proxy-maint
can anyone please merge this PR to close https://bugs.gentoo.org/659070

**Comment by rpolasek** [comment]
how can i speed up the merge process? it takes really so much time :-/

**Comment by lkraav** [comment]
Looks like https://github.com/gentoo/gentoo/pull/7270 got held up by the same license hoopla thing?

**Pull request closed without merging**